### ISSUE TITLE
Epoch and iteration state are lost when loading from checkpoint

### ISSUE THREAD

**Issue by ChrisCummins** [jcjohnson/torch-rnn#72]
When resuming training from a checkpoint, the epoch and iterations counters are reset. This breaks behaviour of the `-max_epochs` setting when training must be split across multiple "sessions".

Example when loading a checkpoint of 7000 iterations:

```
$ th train.lua $OPTIONS -init_from cv/$(ls -t cv | head -n1)
Running with CUDA on GPU 0
Initializing from       cv/checkpoint_7000.t7
Epoch 1.00 / 100, i = 10 / 672300, loss = 0.099103
Epoch 1.00 / 100, i = 20 / 672300, loss = 0.068736
Epoch 1.00 / 100, i = 30 / 672300, loss = 0.071380
```

**Comment by ChrisCummins** [comment]
It seems like it would be straightforward to dump the iteration and epoch variables into the checkpoint JSON files, so I may implement this myself and submit a PR.

**Comment by ChrisCummins** [comment]
Merged in fa666c3991ba34c27fedb7c3eb7f89e0b78a2f63.
